test/integration/cdi/example-driver/app/kubeletplugin.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package app
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	cdiapi "github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	specs "github.com/container-orchestrated-devices/container-device-interface/specs-go"
+	"k8s.io/klog/v2"
+	drapbv1 "k8s.io/kubernetes/pkg/kubelet/apis/dra/v1alpha1"
+	plugin "k8s.io/kubernetes/test/integration/cdi/example-driver/kubeletplugin"
+)
+
+type pluginConfig struct {
+	cdiDir                 string
+	driverName             string
+	draAddress             string
+	pluginRegistrationPath string
+}
+
+type examplePlugin struct {
+	config             pluginConfig
+	resourceHandleKeys []string
+}
+
+var resourceHandleKeys = []string{"some", "another"}
+
+// newExamplePlugin returns an initialized examplePlugin instance.
+func newExampleDriver(config pluginConfig) (*examplePlugin, error) {
+	examplePlugin := &examplePlugin{
+		config: config,
+	}
+	examplePlugin.resourceHandleKeys = makePossibleResourceHandleKeys(resourceHandleKeys)
+	return examplePlugin, nil
+}
+
+// makePossibleResourceHandleKeys returns keys prefixed by "user_" and "admin_".
+func makePossibleResourceHandleKeys(keys []string) []string {
+	var possibleResourceHandkeKeys = []string{}
+	for _, key := range keys {
+		possibleResourceHandkeKeys = append(possibleResourceHandkeKeys, "user_"+key)
+		possibleResourceHandkeKeys = append(possibleResourceHandkeKeys, "admin_"+key)
+	}
+	return possibleResourceHandkeKeys
+}
+
+// getJsonFilePath returns the absolute path where json file is/should be.
+func (ex *examplePlugin) getJsonFilePath(claimUID string) string {
+	return filepath.Join(ex.config.cdiDir, ex.config.driverName+"-"+claimUID+".json")
```

Comment:
I'd suggest to use[ fmt.Sprintf](https://pkg.go.dev/fmt#Sprintf) here instead of string concatenation. I think it would be more readable.